packages/houdini-utils/src/paint/fallback/keyframe.ts

```diff
@@ -0,0 +1,109 @@
+import { lerp } from './lerp';
+import {
+  stepValueToPercentage,
+  timeToNumber,
+  processTimingFunction,
+  parseValue,
+} from './util/css';
+import type {
+  CreateKeyframeAnimationFn,
+  FallbackAnimation,
+  FallbackAnimationValues,
+  EaseMap,
+} from '../../types';
+
+const eases: EaseMap = {
+  linear: lerp,
+};
+
+export const createKeyframeAnimation: CreateKeyframeAnimationFn = ({
+  animations,
+  delay,
+  duration,
+  timingFunction,
+  target,
+}) => {
+  const durations = duration.split(',').map(timeToNumber);
+  const delays = delay.split(',').map(timeToNumber);
+  const timingFunctions = timingFunction.split(',').map(processTimingFunction);
+
+  const anims = [] as FallbackAnimation[];
+  let overallDuration = 0;
+  for (
+    let animationIndex = 0;
+    animationIndex < animations.length;
+    animationIndex++
+  ) {
+    const animation = animations[animationIndex];
+    const steps = Object.entries(animation);
+
+    if (steps.length < 2) {
+      console.error('Animation must have at least 2 steps');
+      return;
+    }
+
+    const delay = delays[animationIndex] ?? 0;
+    const duration = durations[animationIndex] ?? 0;
+    const animDuration = delay + duration;
+
+    if (animDuration > overallDuration) {
+      overallDuration = animDuration;
+    }
+
+    const timingFunction = timingFunctions[animationIndex] ?? 'linear';
+
+    const styles = getComputedStyle(target);
```

Comment:
`target` doesn't change with each iteration and `getComputedStyle()` is a relatively expensive call. We should be able to hoist this up outside the loop to improve perf.